game-core/src/main/java/games/strategy/triplea/ui/MovePanel.java

```diff
@@ -1603,15 +1604,23 @@ public void keyTyped(final KeyEvent e) {}
       @Override
       public void keyPressed(final KeyEvent e) {
         switch (e.getKeyCode()) {
+          case KeyEvent.VK_SPACE:
```

Comment:
I'm not sure if a map would be better here. IMO this is a decent use-case for switch statement.

Reply:
I think switch is ok as long as the amount of cases is low.
In this case it's just a preference thing how many statements you consider sufficient to justify creating a map.

Reply:
Interesting perspective. I would prefer a switch statement when the handling code does not need to be shared, all handling code is defined in one place, and last when the empty case needs explicit handling (either error or otherwise). It's more natural I think IMO to deal with the default vs checking for an empty map result and mapping that to a new action. That kinda distributes handling to multiple places, eg:
```
Map map = new Map...
map.put(...)
map.put(...);

Action action = map.get(key)
if(action == null) {
  // handle error, notice that the error handling here is defined on its own, apart from the map actions above
} else {
  action.doSomething();
}

``` 

I've never really considered switch cases as a criteria for if to choose a map. Thinking to some extreme cases, I think I've seen really big switch statements that had lots of cases. My personal 2 cents, I think it's a bigger factor how much handling should be in each case statement. In essence the cases should be short and sweet. If you there is a lot of code per case, it likely should be extracted to methods. If extracting many cases to many methods, in that case a map is probably better. EG:
```
// this is probably better as a map
switch(var) {
  case 1:
      extractedMethod1();
      break;
   case 2:
      extractedMethod2();
     break;
  :
  :
}
```